server/item/component/item.go

```diff
@@ -0,0 +1,96 @@
+package component
+
+import (
+	"github.com/df-mc/dragonfly/server/item"
+	"github.com/df-mc/dragonfly/server/item/armour"
+	"github.com/df-mc/dragonfly/server/world"
+	"strings"
+)
+
+// FromItem returns all the components of the given custom item. If the item has no components, the second return
+// value is false.
+func FromItem(it world.CustomItem) (map[string]interface{}, bool) {
```

Comment:
Consider moving this somewhere within the internal package, this doesn't look like something we'd want to expose to the user.